### ISSUE TITLE
Unclear formula 6

### ISSUE THREAD

**Issue by artsobolev** [aiha-lab/TSLD#1]
Could you please clarify the formula 6? I think you've used some non-standard notation because the way I see it,

* Pᵀ is a matrix (since L_logit in (5) is presumably a scalar)
* Pᵀₙᵢ is a scalar
* yₙ is a one-hot vector, yₙᵢ is a scalar, too
* Hence the whole argument of the softmax is a scalar, and the point-wise multiplication doesn't make sense either.

Clearly, something's wrong, but I'm not sure how I'm supposed to read this formula. Could you please clarify it?

<img width="684" alt="f6" src="https://github.com/aiha-lab/TSLD/assets/434122/7b1f0e7e-d58e-4a69-8162-980b318ae8da">

**Comment by MarsJacobs** [comment]
Thank you for your valuable comments regarding the formula for TSLD method very kindly. As you rightly pointed out, there seems to be confusion due to the use of non-standard notation, particularly regarding the the operation of softmax and point-wise multiplication. Let me provide a clearer explanation:

**1) Calculation of Per-Token Cross Entropy Loss for the Teacher Model ($L^T_n$)**:
Firstly, we calculate the cross entropy loss for each token $n$ of the teacher model ($L^T_n$). This is done by comparing the teacher model's logits with the ground truth labels($y_{n,i}$). The loss for each token is calculated as follows:
<img width="200" alt="step_1_Teahcer_Loss" src="https://github.com/aiha-lab/TSLD/assets/54992207/68171fa6-33fd-4987-8875-33a71f0d232b">


**2) Application of the Softmax Function**:
The softmax function is then applied to the vector of per-token cross entropy losses ($L^T_n$) to calculate a scaling factor for each token. This is given by (note that $T$ is temperature value in softmax):
<img width="250" alt="step_2_per_token_loss" src="https://github.com/aiha-lab/TSLD/assets/54992207/fd9d6ca3-e6ee-441f-99f8-d6e15ec0e2c4">

**3) Calculation of the Total Loss ($L_{tsld}$)**:
Finally, the total loss $L_{tsld}$ is calculated by applying these scaling factors (Token_Scale_n) to the cross entropy loss between the teacher and student models for each token. The formula for the total loss is:
<img width="450" alt="step_3_tsld_loss" src="https://github.com/aiha-lab/TSLD/assets/54992207/ea10aa12-8142-405a-8c99-aecbf7c7fce4">

Combining these three steps, the formula for $L_{tsld}$ can be represented as follows:
<img width="550" alt="스크린샷 2023-11-18 오후 11 27 12" src="https://github.com/aiha-lab/TSLD/assets/54992207/8ee309fc-7909-453a-9d10-d843406de726">
<img width="600" alt="스크린샷 2023-11-18 오후 10 28 57" src="https://github.com/aiha-lab/TSLD/assets/54992207/06f31d46-5022-4242-a649-6f730e02a35d">

In this formulation, "softmax(...)\_n" indicates that the scaling factor for the $n_{th}$ token, as derived from the softmax output, is multiplied with the cross entropy loss of the $n_{th}$ token. This token-scaling ensures that each token's logit distillation is scaled dynamically based on the cross-entropy loss of the teacher model's predictions for that token.

I am grateful for your constructive feedback. Should there be any further aspects that seem unclear or could benefit from additional refinement, I welcome your suggestions.

**Comment by MarsJacobs** [comment]
Thanks you again for your helpful comments, I've updated the equation for TSLD and released an update on Arxiv. I'd appreciate it if you could check it out. https://arxiv.org/abs/2308.06744

The modified equation is as follows:

<img width="951" alt="스크린샷 2023-12-06 오전 12 15 25" src="https://github.com/aiha-lab/TSLD/assets/54992207/106b466e-752d-4a84-a5bb-63f350aeac59">

Once again, thank you for leaving such a helpful and kind comment on a crucial part of the paper. If you have any further questions or points of discussion, please feel free to reach out at any time.
